adapter/internal/notifier/deployment_notifier.go

```diff
@@ -82,3 +83,51 @@ func SendRevisionUpdate(deployedRevisionList []*DeployedAPIRevision) {
 		}
 	}
 }
+
+// SendRevisionUndeploy - send the undeployed revision information to control plane
+func SendRevisionUndeploy(apiUUID string, revisionUUID string, environment string) {
+	conf, _ := config.ReadConfigs()
+	cpConfigs := conf.ControlPlane
+
+	revisionEP := cpConfigs.ServiceURL
+	if strings.HasSuffix(revisionEP, "/") {
+		revisionEP += unDeployedRevisionEP
+	} else {
+		revisionEP += "/" + unDeployedRevisionEP
+	}
+
+	if apiUUID == "" || revisionUUID == "" || environment == "" {
+		return
+	}
+
+	removedRevision := UnDeployedAPIRevision{
+		APIUUID:      apiUUID,
+		RevisionUUID: revisionUUID,
+		Environment:  environment,
+	}
+
+	jsonValue, _ := json.Marshal(removedRevision)
+	basicAuth := "Basic " + auth.GetBasicAuth(cpConfigs.Username, cpConfigs.Password)
+	retries := 0
+	for retries < 3 {
+		retries++
+		req, _ := http.NewRequest("POST", revisionEP, bytes.NewBuffer(jsonValue))
+		req.Header.Set("Authorization", basicAuth)
```

Comment:
isn't there any constants used for these (while introducing deployRevision Update)?

Reply:
there wasn't . better to have. Will fix.

Reply:
Fixed via : https://github.com/wso2/product-microgateway/pull/2503/commits/b551edd52195244f99b9c10e471d9e431125a7c9